### DISCUSSION TITLE
Detect regions with different backgrounds

### DISCUSSION THREAD

**Discussion by maiiabocharova** [pymupdf/PyMuPDF#1687] [Looking for help] [answered]
Hello, suppose there is a PDF page containing a column with some background color different from the main white background.
I am wondering is there a way to extract the coordinates of the regions based only on background colors using PyMuPDF? I know how to do this using PIL, but maybe there is some better and easier way without conversion, counting colors and determining the regions using some additional logic.

**Comment by JorjMcKie** [comment, marked as answer]
Hello, there is method `page.get_drawings()` which returns lines, rectangles, curves together with any fill colors.
So you can look for a rectangle in that list of dictionaries. In your case it should indeed have a fill color and be large enough, etc.
Then take that rect as the clip parameter in `page.get_text(option, clip=rect, ...more args)`.

This are only slightly different if the background was not created by PDF drawings but by embedded images (which is rare).

**Reply by maiiabocharova** [reply]
Thank you a lot! It solved the issue, indeed much easier compared to doing in with PIL

**Reply by JorjMcKie** [reply]
Wow, that was quick!
